### PR TITLE
[Publisher][Design Update][Metric Settings] Update layout of Metric Definitions page (3/3)

### LINKED ISSUE #1167: [Publisher][Design Updates] Update Metric Settings layout

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Update the layout of the Metric Definitions page to match the [Figma designs](https://www.figma.com/file/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?type=design&node-id=9412%3A1321&mode=design&t=ZZyWrKU782rw1044-1) [note, the character limit counter on the context input will not be implemented - not worth the added complexity at this point].

The changes are mostly styling changes - no-op with regards to functionality.

Demo: https://mahmoudtest---publisher-web-b47yvyxs3q-uc.a.run.app/

## Related issues

Closes #1167

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by helperbot-recidiviz** [comment]
@mxosman successfully triggered a playtest deployment. Full deployment usually takes 5 minutes. Your playtest link is https://mahmoudtest---publisher-web-b47yvyxs3q-uc.a.run.app/

File: publisher/src/components/MetricsConfiguration/MetricDefinitions.tsx [hunk 1]
```diff
@@ -202,22 +183,24 @@ function MetricDefinitions() {
...
-                      <span>View / Edit</span>
+                      <Styled.EditButton
+                        onClick={() => {
+                          setActiveDisaggregationKey(disaggregationKey);
+                          setActiveDimensionKey(key);
+                          setIsSettingsModalOpen(true);
+                        }}
+                      >
+                        Edit
+                      </Styled.EditButton>
```
Comment (mxosman) [inline publisher/src/components/MetricsConfiguration/MetricDefinitions.tsx:200]: Isolating the click to just the Edit button and not the entire block.

**Comment by helperbot-recidiviz** [comment]
@mxosman successfully triggered a playtest deployment. Full deployment usually takes 5 minutes. Your playtest link is https://mahmoudtest---publisher-web-b47yvyxs3q-uc.a.run.app/

**Comment by helperbot-recidiviz** [comment]
@mxosman successfully triggered a playtest deployment. Full deployment usually takes 5 minutes. Your playtest link is https://mahmoudtest---publisher-web-b47yvyxs3q-uc.a.run.app/

**Comment by helperbot-recidiviz** [comment]
@mxosman successfully triggered a playtest deployment. Full deployment usually takes 5 minutes. Your playtest link is https://mahmoudtest---publisher-web-b47yvyxs3q-uc.a.run.app/

**Comment by helperbot-recidiviz** [comment]
@mxosman successfully triggered a playtest deployment. Full deployment usually takes 5 minutes. Your playtest link is https://mahmoudtest---publisher-web-b47yvyxs3q-uc.a.run.app/

**Pull request merged**